### PR TITLE
add CMake variable to project builds: "SKBUILD"

### PR DESCRIPTION
**Pull request by opadron**
Adds a new CLI-defined variable to CMake builds that are ran by scikit-build: "SKBUILD".
Set to "TRUE" when so ran.

### CODE REVIEW

File: skbuild/cmaker.py [hunk 1]
```diff
@@ -241,6 +241,7 @@ def configure(self, clargs=(), generator_id=None):
                '-DPYTHON_VERSION_STRING=' + sys.version.split(' ')[0],
                '-DPYTHON_INCLUDE_DIR=' + python_include_dir,
                '-DPYTHON_LIBRARY=' + python_library,
+               '-DSKBUILD=TRUE',
```
Comment (thewtex) [inline skbuild/cmaker.py]: -DSKBUILD:BOOL=TRUE

to ensure the correct type.

**Comment by opadron** [comment]
@scikit-build/developers PTAL

**Comment by jcfr** [comment]
Nice.

Could you squash ? Thanks

**Comment by opadron** [comment]
#60

**Pull request closed without merging**